File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int result = 0;
        for(int i=n; i>0; i-=2){
            if(n%2 != 0) result += i;
            else result += Math.pow(i, 2);
        }
        return result;
    }
}

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.js
function solution(n) {
    let even = 0; 
    let odd = 0; 
    
    for (let i = n; i > 0; i--) {
        if (i % 2 === 0) even += Math.pow(i, 2);
        else odd += i;
    }
    return n % 2 === 0 ? even : odd;
}